pynest/nest/lib/hl_api_helper.py

```diff
@@ -424,63 +454,43 @@ def model_deprecation_warning(model):
         warnings.warn('\n' + text)
 
 
-def turn_off_deprecation_warning(deprecated_func_name):
-    """Turns off deprecation warnings on SLI and Python level for given
-    function.
+class SuppressedDeprecationWarning(object):
```

Comment:
I thought about it a little more. `SuppressedDeprecationWarning` is also needed in the Topology hl_api, so I would have to access it there as a private member; possible, but slightly breaking with the logic of privacy. It is also not an implementation detail that may change at any time and does not require privacy from that perspective. Users should realise that they should use this context with care based on its name and documentation. Thus, I will leave the name as it is after all.